### PR TITLE
Add font-search extension

### PR DESCRIPTION
**Pull request by enocla**
## Description

This extension allows you to search and preview locally installed fonts.

## Screencast

font-search-3
font-search-2
font-search-1

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Could you use something like [useCachedPromise](https://developers.raycast.com/utilities/react-hooks/usecachedpromise) so it's instant shows the list while updating

- Would it make sense to add pinned items?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by Tnixc** [comment]
image

- Added ability to pin fonts
- useCachedPromise
- Sort fonts by default

File: extensions/font-search/src/search-fonts.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+          <Action.CopyToClipboard
+            title="Copy Font Name"
+            content={font.name}
+            onCopy={() => showToast({ title: "Font name copied", style: Toast.Style.Success })}
+          />
+          <Action.ShowInFinder path={font.path} />
+          <Action
+            title={font.isPinned ? "Unpin Font" : "Pin Font"}
+            icon={font.isPinned ? Icon.TackDisabled : Icon.Tack}
+            onAction={onTogglePin}
```
Comment (pernielsentikaer) [inline extensions/font-search/src/search-fonts.tsx:206]:
```suggestion
            onAction={onTogglePin}
            shortcut={Keyboard.Shortcut.Common.Pin}
```

File: extensions/font-search/src/search-fonts.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  ActionPanel,
+  List,
+  Action,
+  Icon,
+  showToast,
+  Toast,
+  getPreferenceValues,
+  Cache,
+  LocalStorage,
+} from "@raycast/api";
```
Comment (pernielsentikaer) [inline extensions/font-search/src/search-fonts.tsx:12]:
```suggestion
import {
  ActionPanel,
  List,
  Action,
  Icon,
  showToast,
  Toast,
  getPreferenceValues,
  Cache,
  LocalStorage,
  Keyboard,
} from "@raycast/api";
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
The speed is now much better, awesome!

Should we add a section with pinned entries as default Raycast does?

Raycast07102024-jE0iITNr png@2x

**Comment by Tnixc** [comment]
Oh yeah that's definitely a good idea. Didn't think of this.

**Comment by pernielsentikaer** [comment]
Is this ready for another review?

**Comment by Tnixc** [comment]
yes

**Comment by pernielsentikaer** [comment]
Some fonts seems to be added twice which gives an error since it's not unique, should there be multiple rows with the same path?

```
15:14:29.841 Warning: Encountered two children with the same key, `/System/Library/Fonts/Supplemental/Songti.ttc`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
    at ray-list-section
    at qa (/Applications/Raycast.app/Contents/Resources/RaycastNodeExtensions_RaycastNodeExtensions.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:20:20459)
    at Lr (/Applications/Raycast.app/Contents/Resources/RaycastNodeExtensions_RaycastNodeExtensions.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:8:3631)
    at ray-list
    at lt (/Applications/Raycast.app/Contents/Resources/RaycastNodeExtensions_RaycastNodeExtensions.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:20:24444)
    at Command (/Users/pernielsentikaer/.config/raycast/extensions/font-search/search-fonts.js:7561:65)
    at _r (/Applications/Raycast.app/Contents/Resources/RaycastNodeExtensions_RaycastNodeExtensions.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:8:2593)
    at ray-navigation-stack
    at Eo (/Applications/Raycast.app/Contents/Resources/RaycastNodeExtensions_RaycastNodeExtensions.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:8:2094)
    at Suspense
    at Nr (/Applications/Raycast.app/Contents/Resources/RaycastNodeExtensions_RaycastNodeExtensions.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:7:2335)
    at ray-root
    at Ii (/Applications/Raycast.app/Contents/Resources/RaycastNodeExtensions_RaycastNodeExtensions.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:8:3103)
```

https://github.com/user-attachments/assets/2de86033-1dcb-4ba7-85aa-b49ff6449e19

**Comment by Tnixc** [comment]
Yes, there would be multiple font names with the same path which are different. I've changed the key to fix this.

**Review by pernielsentikaer — APPROVED** [review]
That better, let's get this released 🔥

**Comment by uncenter** [comment]
Loving this extension, thanks! ❤️

**Pull request merged**